app/src/components/Input/index.js

```diff
@@ -1,45 +1,54 @@
 import React from 'react';
+import _ from 'lodash';
 
 class Input extends React.Component {
   constructor(props) {
     super(props);
-    
+    this.handleKeyPress = this.handleKeyPress.bind(this);
     this.handleChange = this.handleChange.bind(this);
-    const value = this.props.value || '';
-    this.state = { value };
+    this.state = {
+      value: this.props.value ? _.clone(this.props.value) : ''
+    };
   }
 
-  handleChange(element) {
+  handleKeyPress(e) {
+    if (e.charCode === 13) {
+      e.preventDefault();
+      
+      /* TODO: decide if this is a nice feature to have
```

Comment:
Same thing here, let's leave TODOs on dev branches and try to keep master as clean as possible.